pydantic/main.py

```diff
@@ -742,3 +764,26 @@ def validate_model(  # noqa: C901 (ignore complexity)
         raise err
 
     return values, fields_set, None
+
+
+@lru_cache(maxsize=None)
+def _get_parsing_type(type_: Any) -> Any:
+    class ParsingModel(BaseModel):
+        value: type_  # type: ignore
+
+    return ParsingModel
+
+
+T = TypeVar('T')
+
+
+def parse_as_type(obj: Any, type_: T) -> T:
+    model_type = _get_parsing_type(type_)
```

Comment:
This method is pretty simple, I'm not sure:
1. What you're using it for here.
2. what it's wider application is.
3. whether it needs to be included in pydantic, I imagine it would work fine without being including in this package?

Reply:
I could drop it -- it's not necessary for the application I have in mind, it's just that it felt nicely paired with the `dump_as_type` function, and I believe it came up in a separate github issue.

I view it as a nice utility function that I can call from anywhere with any type pydantic supports and get the pydantic-parsed version of the type, without even needing to create a BaseModel or anything.

Besides the clean api, there is also the caching happening in `_get_parsing_type` that ensures high performance.

Based on the usage of pydantic I've seen in the wild from the fastapi gitter and github issues in both of these projects, my guess is that most people won't reproduce this function, since any version of it would make use of relatively uncommon usage of pydantic. The utility of an all-purpose parsing function that could be used anywhere with any type seems self-evident to me, but I've been wrong about plenty of such things before 😄.

Reply:
makes sense, let's keep them.

But as per #481 let's move them to a new module `pydantic.tools`.

Reply:
👍 for the move to a new module; it definitely felt inappropriate in main, but I couldn't get imports to work in utils